django/db/backends/mysql/base.py

```diff
@@ -360,6 +360,8 @@ def check_constraints(self, table_names=None):
                     )
 
     def is_usable(self):
+        if self.connection is None:
+            return False
```

Comment:
why a `try/except` when a `is None` do the job? I don't see any good reason to do it.
but pushing the logic in the base is a good idea for sure.